pkg/apis/pipeline/v1beta1/when_types.go

```diff
@@ -99,8 +101,13 @@ type WhenExpressions []WhenExpression
 // AllowsExecution evaluates an Input's relationship to an array of Values, based on the Operator,
 // to determine whether all the When Expressions are True. If they are all True, the guarded Task is
 // executed, otherwise it is skipped.
-func (wes WhenExpressions) AllowsExecution() bool {
+// If CEL expression exists, AllowsExecution will get the evaluated results from evaluatedCEL and determine
+// if the Task should be skipped.
+func (wes WhenExpressions) AllowsExecution(evaluatedCEL map[string]bool) bool {
 	for _, we := range wes {
+		if we.CEL != "" {
+			return evaluatedCEL[we.CEL]
```

Comment:
is it possible for the key not to be in the map? what happens in that case?

Reply:
That's a great question. In current implementation it is not possible, we will traverse all cel expressions and evaluated them. 
I have thought about adding this check but it is hard to return an error here, all the returned values are bool in the call stack. And that's also the reason I move the evaluation logic upfront